pkg/docker/docker.go

```diff
@@ -274,21 +304,27 @@ func (d *stiDocker) UploadToContainer(src, dest, name string) error {
 		path = dest
 		go func() {
 			defer w.Close()
-			if err := t.StreamDirAsTar(src, dest, w); err != nil {
+			if err := t.StreamDirAsTarWithCallback(src, dest, w, walkFn, modifyInplace); err != nil {
 				glog.V(0).Infof("error: Uploading directory to container failed: %v", err)
 			}
 		}()
 	} else {
 		go func() {
 			defer w.Close()
-			if err := t.StreamFileAsTar(src, filepath.Base(dest), w); err != nil {
+			if err := t.StreamFileAsTarWithCallback(src, filepath.Base(dest), w, walkFn, modifyInplace); err != nil {
 				glog.V(0).Infof("error: Uploading files to container failed: %v", err)
 			}
 		}()
 	}
 	glog.V(3).Infof("Uploading %q to %q ...", src, path)
 	opts := docker.UploadToContainerOptions{Path: path, InputStream: r}
-	return d.client.UploadToContainer(name, opts)
+	return d.client.UploadToContainer(container, opts)
```

Comment:
given that this is how we are uploading the content into the container, do we actually still require a "tar" binary in the image?  We used to require tar to decode the tar stream we sent into the container, but it seems like we're no longer doing it that way?